src/users/users.service.ts

```diff
@@ -33,10 +37,25 @@ export class UsersService {
     user: Partial<UserEntity>,
     password: string,
   ): Promise<UserEntity> {
+    if (user.username.length < 5)
+      throw new BadRequestException('Username must be of minimum 5 characters');
+
+    if (password.length < 8)
+      throw new BadRequestException('Password must be of minimum 8 characters');
+
+    if (password.toLowerCase().includes('password'))
+      throw new BadRequestException(
+        'Password cannot contain the word password itself',
+      );
+
+    const usernameAlreadyExists = await this.getUserByUsername(user.username);
+    if (usernameAlreadyExists)
+      throw new ConflictException('This username is already taken!');
+
     const newUser = await this.userRepo.save(user);
-    // TODO: check for username/email existing and throw proper error
-    // TODO: check username min length 5 chars
-    // TODO: check password min length 8 chars
+    // TODO: check for username/email existing and throw proper error - ✅
```

Comment:
remove the TODOs from here 

Ctrl-R `todo` is the best way to find out pending stuff in a project. The comments should not remain TODO if the work is DONE already.

Reply:
Noted. I've removed the TODOs in `users.service.ts`.